src/Servers/HttpSys/test/FunctionalTests/Utilities.cs

```diff
@@ -148,17 +151,38 @@ internal static IServer CreateDynamicHttpServer(string basePath, out string root
             throw new Exception("Failed to locate a free port.");
         }
 
-        internal static IServer CreateHttpsServer(RequestDelegate app)
+        internal static IServer CreateDynamicHttpsServer(out string baseAddress, RequestDelegate app)
         {
-            return CreateServer("https", "localhost", 9090, string.Empty, app);
+            return CreateDynamicHttpsServer("/", out var root, out baseAddress, options => { }, app);
         }
 
-        internal static IServer CreateServer(string scheme, string host, int port, string path, RequestDelegate app)
+        internal static IServer CreateDynamicHttpsServer(string basePath, out string root, out string baseAddress, Action<HttpSysOptions> configureOptions, RequestDelegate app)
         {
-            var server = CreatePump();
-            server.Features.Get<IServerAddressesFeature>().Addresses.Add(UrlPrefix.Create(scheme, host, port, path).ToString());
-            server.StartAsync(new DummyApplication(app), CancellationToken.None).Wait();
-            return server;
+            lock (PortLock)
+            {
+                while (NextHttpsPort < MaxHttpsPort)
+                {
+
+                    var port = NextHttpsPort++;
+                    var prefix = UrlPrefix.Create("https", "localhost", port, basePath);
+                    root = prefix.Scheme + "://" + prefix.Host + ":" + prefix.Port;
+                    baseAddress = prefix.ToString();
+
+                    var server = CreatePump();
+                    server.Features.Get<IServerAddressesFeature>().Addresses.Add(baseAddress);
+                    configureOptions(server.Listener.Options);
+                    try
+                    {
+                        server.StartAsync(new DummyApplication(app), CancellationToken.None).Wait();
+                        return server;
+                    }
+                    catch (HttpSysException)
+                    {
```

Comment:
Log informational here?

Reply:
No, we're expecting port conflict errors, it's Debug at best. These are also not Logged tests, there's nowhere to log to.